Library/Homebrew/dev-cmd/unbottled.rb

```diff
@@ -186,6 +186,15 @@ def output_unbottled(formulae, deps_hash, noun, hash, any_named_args)
       end
 
       requirements = f.recursive_requirements
+
+      # On Linux, a LinuxRequirement could be added to requirements due to any
+      # Linux-specific Linux-only dependents. This will conflict with a
+      # MacOSRequirement, and incorrectly output that the formula requires
+      # Linux for a macOS bottle tag. Such a LinuxRequirement must be removed.
+      if requirements.any?(MacOSRequirement) && requirements.any?(LinuxRequirement)
+        requirements.reject! { |r| r.is_a?(LinuxRequirement) }
+      end
```

Comment:
This should not happen unless `on_linux` blocks are being evalauted, no?

Sounds like something's gone wrong if those blocks are being evaluated.

Reply:
Yeah seems like those blocks are being evaluated because the formula is loaded as it normally would be on Linux, despite specifying a macOS bottle tag. So the command output could be incorrect because we recursively check potentially Linux-only dependencies for a macOS tag.

Reply:
Hmm odd given we have `Homebrew::SimulateSystem` wrapping all this. Seems like a bad bug if so given we rely on that for audits and readall checks.

Reply:
Actually is this with or without `HOMEBREW_NO_INSTALL_FROM_API` or is it no difference?

Reply:
Ah this is using the API on Linux. Here's some hopefully relevant output:
```
linuxbrew@0e6abcd7b4ed:~/.linuxbrew/Homebrew/Library/Homebrew$ brew unbottled --tag big_sur tart
==> Populating dependency tree...
==> :big_sur bottle status
tart: requires Linux
linuxbrew@0e6abcd7b4ed:~/.linuxbrew/Homebrew/Library/Homebrew$ HOMEBREW_NO_INSTALL_FROM_API=1 brew unbottled --tag big_sur tart
==> Populating dependency tree...
==> :big_sur bottle status
Error: undefined method `latest_version' for OS::Mac::Xcode:Module
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/unbottled.rb:208:in `block (2 levels) in output_unbottled'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/set.rb:338:in `each_key'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/set.rb:338:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/delegate.rb:83:in `all?'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/lib/ruby/2.6.0/delegate.rb:83:in `method_missing'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/unbottled.rb:199:in `block in output_unbottled'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/unbottled.rb:180:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/unbottled.rb:180:in `output_unbottled'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/unbottled.rb:93:in `unbottled'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:93:in `<main>'
```
So it seems like that `LinuxRequirement` is added only for the API. To confirm, I tried printing `f.recursive_dependencies` with and without the API, and it prints Linux-only dependencies for a simulated macOS only on the API.

Reply:
Yes, simulation is not supported with the API yet.

Reply:
This seems to have worked, thanks Bo!